internal/controllers/machineset/machineset_controller.go

```diff
@@ -969,12 +994,47 @@ func (r *Reconciler) getMachineNode(ctx context.Context, cluster *clusterv1.Clus
 
 func (r *Reconciler) reconcileUnhealthyMachines(ctx context.Context, cluster *clusterv1.Cluster, ms *clusterv1.MachineSet, filteredMachines []*clusterv1.Machine) (ctrl.Result, error) {
 	log := ctrl.LoggerFrom(ctx)
+
+	// Calculate how many in flight machines we should remediate.
+	// By default, we allow all machines to be remediated at the same time.
+	maxInFlight := len(filteredMachines)
+
+	// If the MachineSet is part of a MachineDeployment, only allow remediations if
+	// it's the desired revision.
+	if util.HasOwner(ms.GetOwnerReferences(), clusterv1.GroupVersion.String(), []string{"MachineDeployment"}) {
+		owner, err := r.getOwnerMachineDeployment(ctx, ms)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if owner.Annotations != nil && ms.Annotations != nil && owner.Annotations[clusterv1.RevisionAnnotation] != ms.Annotations[clusterv1.RevisionAnnotation] {
+			// MachineSet is part of a MachineDeployment but isn't the current revision, no remediations allowed.
+			return ctrl.Result{}, nil
```

Comment:
q: should we update machine's OwnerRemediatedCondition with a message that explains that remediation is not going to happen because the machine is going to be deleted due to rollout?

Reply:
Both here and below, I was trying to avoid further patches to Machines during operations; the condition stays false until remediation can take place, which sounded a good compromise